spec/jobs/jobs/prometheus_customer_auth_stats_spec.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe Jobs::PrometheusCustomerAuthStats do
+  subject do
+    job.call
+  end
+
+  let(:job) { described_class.new(double) }
+
+  let(:account1) { FactoryBot.create(:account, external_id: 1) }
+  let(:account2) { FactoryBot.create(:account, external_id: 2) }
+  let(:account3) { FactoryBot.create(:account, external_id: 3) }
+
+  let(:customer_auth1) { FactoryBot.create(:customers_auth, account: account1, external_id: 4, external_type: 'test1') }
+  let(:customer_auth2) { FactoryBot.create(:customers_auth, account: account2, external_id: 5, external_type: 'test2') }
+  let(:customer_auth3) { FactoryBot.create(:customers_auth, account: account3, external_id: 6, external_type: 'test3') }
```

Comment:
let's set `external_type: nil` for customer_auth2

let's set `external_id: nil, external_type: nil` for customer_auth3